### PR TITLE
Compatibility for NOAA and IRIDL data sets

### LINKED ISSUE #293: Data with coordinate variables (lat, lon, etc.) with names not included in internal_names
This may be two separate questions, but longitudes are often treated differently across models. Specifically:
1) Is there a way to define `alt_names` for a coordinate so that we can deal with something which is `X`, `lon`, or `longitude?
2) Is there a way to join longitudes that go from 0 to 360 with longitudes that go from -180 to 180 so that the outputs are standardized and everything works well with regions?
This seems like something which would go in the `model` creation

### PR DESCRIPTION
**Pull request by jdossgollin**
Closes #293 and #299.
This PR makes two changes which make aospy compatible with a greater range of data sets, specifically including those from the [IRI Data Library](http://iridl.ldeo.columbia.edu/) and NOAA data sets such as the [NCEP-NCAR Reanalysis](https://www.esrl.noaa.gov/psd/data/gridded/data.ncep.reanalysis.html). This compatibility is expanded through (1) the addition of several new names for `GRID_ATTRS` as discussed in #293 (for `X`, `Y`, `P`, and #299 (for `nbnds`); `land` was not discussed in a previous issue but is also a relevant use case without likelihood of confusion, and (2) a modification of `ensure_time_avg_has_cf_metadata` to avoid the assumption that `BOUNDS_STR` is a coordinate of `time_weights` as discussed in #299.

### CODE REVIEW

File: docs/whats-new.rst [hunk 1]
```diff
@@ -37,6 +37,12 @@ Breaking Changes
 - Deprecate ``func_input_dtype`` argument to ``Var`` (fixes
   :issue:`281` via :pull:`282`).  By `Spencer Hill
   <https://github.com/spencerahill>`_.
+- Improve compatibility for data following IRIDL conventions or NOAA
+  data formats. Specifically, several alternate names are defined in
+  `GRID_ATTRS`, while there is no longer an assumption that
```
Comment (spencerkclark) [inline docs/whats-new.rst]: In restructured text, which is what we use to write the documentation, [inline literals](http://docutils.sourceforge.net/docs/ref/rst/restructuredtext.html#inline-literals) are specified using double back-ticks, e.g.:
```
``GRID_ATTRS``
```
So add a couple more back-ticks here for `GRID_ATTRS` as well as some for `BOUNDS_STR` and `time_weights`.

This is slightly different than GitHub Markdown, so it takes some time getting used to...

**Review by spencerkclark — COMMENTED** [review]
This looks really good @jdossgollin; I just have one minor suggestion regarding the formatting of the what's new entry.

File: docs/whats-new.rst (hunk 1 above)
Comment (jdossgollin) [reply docs/whats-new.rst]: whoops! I'm on it. 
@spencerkclark is the procedure to close this PR, push another commit to my branch, and then submit a new PR?

Comment (spencerkclark) [reply docs/whats-new.rst]: No need to close the PR!  Just push a new commit to the branch and it will show up here.

Comment (jdossgollin) [reply docs/whats-new.rst]: You mean like where it says 

> "Add more commits by pushing to the noaa-iridl-compat branch on jdossgollin/aospy."

down below? 😄

Comment (spencerkclark) [reply docs/whats-new.rst]: Perfect!

**Review by spencerkclark — APPROVED** [review]

**Comment by spencerahill** [comment]
Thanks much @jdossgollin ! Congrats on your first aospy PR, and sorry again for those aspects of the procedure that we accidentally made pretty confusing.

**Comment by jdossgollin** [comment]
Thanks for the patience and guidance @spencerahill  and @spencerkclark !

**Comment by spencerahill** [comment]
No problem at all.  Looking forward to additional future contributions from you :)

**Pull request merged**